homeassistant/core.py

```diff
@@ -2025,7 +2038,9 @@ def async_remove(self, entity_id: str, context: Context | None = None) -> bool:
         old_state.expire()
         self._bus._async_fire(  # pylint: disable=protected-access
             EVENT_STATE_CHANGED,
-            {"entity_id": entity_id, "old_state": old_state, "new_state": None},
+            EventStateChangedData(
+                entity_id=entity_id, old_state=old_state, new_state=None
+            ),
             context=context,
         )
```

Comment:
This option makes the code equally clear to the reader at no performance cost:

```py
        state_changed_data: EventStateChangedData = {"entity_id": entity_id, "old_state": old_state, "new_state": None}
        self._bus._async_fire(  # pylint: disable=protected-access
            EVENT_STATE_CHANGED,
            state_changed_data,
            context=context,
        )
```

Reply:
There is a tiny performance cost even with this option but I agree that for `EVENT_STATE_CHANGED` this is better.

```py
from timeit import timeit

setup2 = """\
def func(var):
    return None
"""
print(f"{timeit('d = {"action": "create"}; func(d)', setup=setup2)}")
print(f"{timeit('func(dict(action="create"))', setup=setup2)}")
print(f"{timeit('func({"action": "create"})', setup=setup2)}")
```

```
0.038815999985672534
0.05378125002607703
0.03651975002139807
```